### PR TITLE
Move it over!

### PR DESCRIPTION
**Pull request by drapergeek**
This is just a quick pass at moving over the existing implementation and tests.

What's here:
- Implementation
- Basic test coverage
- Basic examples in the dummy app

What's missing that will come in followup PRs:
- The test helpers. I want to move those over, if you're using this addon, that will be a nice feature.
- DOCUMENTATION!
- The standard thoughtbot OSS stuff.

### CODE REVIEW

**Comment by seanpdoyle** [comment]
A GIF in this PR (as well as in the future README) would be nice.

Do we have a working example somewhere?

**Comment by drapergeek** [comment]
> A GIF in this PR (as well as in the future README) would be nice.

Agreed, that would be quite nice.

> Do we have a working example somewhere?

In the dummy app here I've wired up a basic implementation as an example.

File: addon/templates/components/inline-edit-field.hbs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  <form class="editable is-editing"
+    {{action "stopEditing" model on="submit"}}>
+
+    {{#if (hasBlock "inverse")}}
+      {{yield to="inverse"}}
+    {{else}}
+      {{input value=value}}
+    {{/if}}
+
+    <div class="save-and-cancel">
```
Comment (BlakeWilliams) [inline addon/templates/components/inline-edit-field.hbs:11]: Should `yield` be all or nothing? I assume if we want to be able to customize the input we'd also want to be able to customize the submit buttons?

File: app/components/inline-edit-field.js [hunk 2]
```diff
@@ -0,0 +1 @@
+export { default } from 'inline-edit-field/components/inline-edit-field';
```
Comment (MattMSumner) [inline app/components/inline-edit-field.js]: This must have been generated? Missing newline.

**Comment by drapergeek** [comment]
inlineedit

File: app/components/inline-edit-field.js (hunk 2 above)
Comment (drapergeek) [inline app/components/inline-edit-field.js]: Exactly correct.

File: tests/dummy/app/routes/main.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import Ember from 'ember';
+
+export default Ember.Route.extend({
+  model: function() {
+    return {
+      name: 'Ralph the Robot',
+      bio: "Just a small town girl. Livin' in a lonely world'",
```
Comment (MattMSumner) [inline tests/dummy/app/routes/main.js:7]: :+1:

**Comment by seanpdoyle** [comment]
What do you think about changing the `<a href="#">` tags to `<button>`s?

If it's the clickable-text we're after, buttons can be styled to look like links (right?).

File: addon/templates/components/inline-edit-field.hbs (hunk 1 above)
Comment (drapergeek) [inline addon/templates/components/inline-edit-field.hbs:11]: I haven't had that be the case so far. Normally it's just the fact that you need a different type of input.

I'd like to explore solutions for customizing those buttons moving forward though.

Comment (BlakeWilliams) [inline addon/templates/components/inline-edit-field.hbs:11]: :+1:

**Comment by drapergeek** [comment]
> What do you think about changing the <a href="#"> tags to <button>s?

Yes very much.

**Comment by MattMSumner** [comment]
:+1: to @seanpdoyle's comment regarding links vs buttons. A button seems the right choice.

File: addon/templates/components/inline-edit-field.hbs (hunk 1 above)
Comment (drapergeek) [inline addon/templates/components/inline-edit-field.hbs:11]: Also, you can customize the text of those buttons already by passing in the values.

**Pull request merged**